File: SummerLandBackend/Models/Product.cs
namespace SummerLandBackend.Models;

public class Product
{
    public int Id { get; set; }

    public string ModelNumber { get; set; } = string.Empty;

    public string Name { get; set; } = string.Empty;

    public int CategoryId { get; set; }

    public Category Category { get; set; } = null!;

    public ICollection<ProductVariant> Variants { get; set; }
        = new List<ProductVariant>();
}

File: summerland/lib/models/product.dart
class Product {
  final int id;
  final String modelNumber;
  final String name;
  final int categoryId;
  final String categoryName;

  Product({
    required this.id,
    required this.modelNumber,
    required this.name,
    required this.categoryId,
    required this.categoryName,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'],
      modelNumber: json['modelNumber'],
      name: json['name'],
      categoryId: json['categoryId'],
      categoryName: json['categoryName'],
    );
  }
}